kernel/src/fs/procfs/version.rs

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module offers `/proc/version` file support, which provides
+//! information about the kernel version.
+//!
+//! Reference: <https://man7.org/linux/man-pages/man5/proc_version.5.html>
+
+use alloc::format;
+
+use aster_util::printer::VmPrinter;
+
+use crate::{
+    fs::{
+        procfs::template::{FileOps, ProcFileBuilder},
+        utils::{mkmod, Inode},
+    },
+    net::UtsNamespace,
+    prelude::*,
+};
+
+pub struct VersionFileOps;
+
+impl VersionFileOps {
+    pub fn new_inode(parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self, mkmod!(a+r))
+            .parent(parent)
+            .build()
+            .unwrap()
+    }
+
+    fn collect_version() -> Result<String> {
+        // Get UTS namespace information from the init namespace
+        let uts_name = UtsNamespace::get_init_singleton().uts_name();
```

Comment:
Instead of using information from the init UTS namespace, I think you should get information from the UTS namespace of current thread. The code may look like: 

```rust
let current_thread = current_thread!();
let posix_thread = current_thread.as_posix_thread().unwrap();
let ns_proxy = posix_thread.ns_proxy().lock();
let Some(ns_proxy) = ns_proxy.as_ref() else {
    // Return some error here like ESRCH?
};
let uts_ns = ns_proxy.uts_ns();
......
```
You can refer to the Linux implementation of `uts_name`.
https://elixir.bootlin.com/linux/v6.0.9/source/fs/proc/version.c#L12